### PR TITLE
ci: Pack&send NSC workspace to Artifactory 

### PR DESCRIPTION
**Pull request by jaci-nordic**
Triggered by adding tag. It packs whole repository and sends it to artifactory.

### CODE REVIEW

File: .github/workflows/zip-upload.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - '*'
+
+jobs:
+  zip-and-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          path: sdk-nrf
```
Comment (thst-nordic) [inline .github/workflows/zip-upload.yml:14]: this needs to be `nrf`

File: .github/workflows/zip-upload.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  push:
+    tags:
+      - '*'
+
+jobs:
+  zip-and-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (DatGizmo) [inline .github/workflows/zip-upload.yml:12]: We should be using the reusable action Kari created for then checkout + west update:
https://github.com/NordicBuilder/action-checkout-west-update

**Review by DatGizmo — CHANGES_REQUESTED** [review]

File: .github/workflows/zip-upload.yml (hunk 1 above)
Comment (jaci-nordic) [reply .github/workflows/zip-upload.yml]: Done

File: .github/workflows/zip-upload.yml (hunk 2 above)
Comment (jaci-nordic) [reply .github/workflows/zip-upload.yml]: Done

**Review by DatGizmo — APPROVED** [review]

**Review by thst-nordic — CHANGES_REQUESTED** [review]
need entire workspace, not just sdk-nrf.

File: .github/workflows/zip-upload.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        run: >
+          tar -C ./workspace -cvf src.tar.gz .
+
+      - name: Upload to Artifactory
+        env:
+          URL: https://eu.files.nordicsemi.com/artifactory/ncs-src-mirror
+        run: >
+          curl -u ${{ secrets.COM_NORDICSEMI_FILES_BURAN_CI }}
+          -T src.tar.gz
+          ${URL}/${{ github.ref_name }}/sdk-nrf-source.tar.gz
```
Comment (thst-nordic) [inline .github/workflows/zip-upload.yml:32]: `sdk-nrf-source.tar.gz` can be removed.   keep filename same `src.tar.gz`

**Review by thst-nordic — APPROVED** [review]

**Pull request merged**